package.json

```diff
@@ -1,13 +1,34 @@
 {
   "name": "fluent-javascript",
-  "version": "0.0.1",
   "private": true,
+  "workspaces": [
+    "./fluent-dedent",
+    "./fluent-bundle",
+    "./fluent-sequence",
+    "./fluent-dom",
+    "./fluent-langneg",
+    "./fluent-react",
+    "./fluent-syntax",
+    "./fluent-gecko"
+  ],
+  "scripts": {
+    "predist": "npm run clean",
+    "dist": "npm run build --workspaces",
+    "postdist": "npm run lint && npm run test && npm run docs --workspaces",
+    "clean": "git clean -fdxe node_modules fluent-*/{coverage,dist,esm,index.js} html",
+    "deploy-html": "gh-pages -d html",
+    "lint": "eslint --max-warnings 0 'fluent-*/{src,test}/**/*.{js,ts}'",
+    "test:common": "c8 mocha 'fluent-*/test/*_test.js'",
+    "test": "npm run test:common && npm test -w fluent-dom && npm test -w fluent-react"
```

Comment:
Issue: This fails for me locally:
I'm running npm 7.7.6.

```
~/dev/fluent.js on fluent.js/pr/569 at 09:21:31
➤ npm test

> test
> npm run test:common && npm test -w fluent-dom && npm test -w fluent-react


> test:common
> c8 mocha 'fluent-*/test/*_test.js'


Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/greg/dev/fluent.js/fluent-langneg/esm/accepted_languages.js' imported from /Users/greg/dev/fluent.js/fluent-langneg/test/headers_test.js
    at new NodeError (node:internal/errors:329:5)
    at finalizeResolution (node:internal/modules/esm/resolve:323:11)
    at moduleResolve (node:internal/modules/esm/resolve:758:10)
    at Loader.defaultResolve [as _resolve] (node:internal/modules/esm/resolve:869:11)
    at Loader.resolve (node:internal/modules/esm/loader:88:40)
    at Loader.getModuleJob (node:internal/modules/esm/loader:241:28)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:57:40)
    at link (node:internal/modules/esm/module_job:56:36)
-----------------------------|---------|----------|---------|---------|-----------------------------
File                         | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
-----------------------------|---------|----------|---------|---------|-----------------------------
All files                    |   36.79 |    99.15 |       0 |   36.79 |
 fluent-bundle/esm           |   33.02 |      100 |       0 |   33.02 |
  builtins.js                |   70.55 |      100 |       0 |   70.55 | 14-22,62-78,130-146
  bundle.js                  |   64.38 |      100 |       0 |   64.38 | ...-59,70-71,91-112,141-159
  resolver.js                |   16.48 |      100 |       0 |   16.48 | ...,213-226,229-264,267-273
  resource.js                |   11.44 |      100 |       0 |   11.44 | 47-430,434-436
  scope.js                   |   15.15 |      100 |       0 |   15.15 | 3-14,16-20,22-32
  types.js                   |   70.37 |      100 |       0 |   70.37 | ...58-60,65-73,92-94,99-107
 fluent-bundle/test          |   37.74 |      100 |       0 |   37.74 |
  arguments_test.js          |   45.42 |      100 |       0 |   45.42 | ...,218-222,231-232,236-259
  attributes_test.js         |   41.21 |      100 |     100 |   41.21 | ...,173-182,186-189,193-196
  bomb_test.js               |   49.02 |      100 |     100 |   49.02 | 13,18-31,35-39,43-48
  constructor_test.js        |   42.31 |      100 |     100 |   42.31 | 12,16,20-33,37-50
  context_test.js            |   43.66 |      100 |     100 |   43.66 | ...,120-124,128-132,136-138
  errors_test.js             |   43.48 |      100 |     100 |   43.48 | 13-16,20-35,39-44
  functions_builtin_test.js  |   11.36 |      100 |     100 |   11.36 | ...,477-509,513-548,552-587
  functions_runtime_test.js  |   57.45 |      100 |     100 |   57.45 | 13,18-28,32-35,41-44
  functions_test.js          |   48.15 |      100 |     100 |   48.15 | ...9-82,88-91,95-98,102-105
  isolating_test.js          |   52.13 |      100 |     100 |   52.13 | ...-59,63-69,77-81,85,89-92
  literals_test.js           |      35 |      100 |     100 |      35 | ...,30-39,43-52,56-65,69-78
  macros_test.js             |   39.53 |      100 |     100 |   39.53 | ...,410-413,417-420,424-427
  object_prototype_test.js   |   46.86 |      100 |     100 |   46.86 | ...,217-221,225-229,233-236
  parser_reference_test.js   |    87.1 |      100 |     100 |    87.1 | 25-28
  parser_structure_test.js   |    87.1 |      100 |     100 |    87.1 | 25-28
  patterns_test.js           |   43.23 |      100 |     100 |   43.23 | ...,196-212,216-219,223-226
  primitives_test.js         |   40.68 |      100 |     100 |   40.68 | ...,156-159,163-166,170-173
  select_expressions_test.js |   33.57 |      100 |     100 |   33.57 | ...,101-110,114-123,127-136
  transform_test.js          |   46.77 |      100 |     100 |   46.77 | ...,34-39,43-46,50-53,57-60
  values_format_test.js      |   47.44 |      100 |     100 |   47.44 | ...,50-53,57-60,64-67,71-74
  values_ref_test.js         |   32.65 |      100 |     100 |   32.65 | ...,115-124,128-134,138-144
 fluent-dedent               |   43.64 |       75 |       0 |   43.64 |
  index.js                   |   43.64 |       75 |       0 |   43.64 | 4,22-51
 fluent-dedent/esm           |   31.82 |      100 |       0 |   31.82 |
  index.js                   |   31.82 |      100 |       0 |   31.82 | 15-44
 fluent-dedent/test          |   40.68 |      100 |     100 |   40.68 |
  blank_test.js              |    37.5 |      100 |     100 |    37.5 | ...,29-35,39-44,48-53,57-62
  content_test.js            |    37.5 |      100 |     100 |    37.5 | ...,18-24,28-34,38-44,48-54
  eol_test.js                |   47.37 |      100 |     100 |   47.37 | 8-11,15-19,23-27,31-36
  interpolation_test.js      |   42.86 |      100 |     100 |   42.86 | 8-13,17-22,26-33
  mixed_test.js              |   45.45 |      100 |     100 |   45.45 | 8-13,17-22,26-31
  tabs_test.js               |   39.13 |      100 |     100 |   39.13 | ...,35-40,44-49,53-58,62-67
-----------------------------|---------|----------|---------|---------|-----------------------------
```

Reply:
After running `npm clean` I get a different error:

```
➤ npm test

> test
> npm run test:common && npm test -w fluent-dom && npm test -w fluent-react


> test:common
> c8 mocha 'fluent-*/test/*_test.js'


Error [ERR_MODULE_NOT_FOUND]: Cannot find package '/Users/greg/dev/fluent.js/node_modules/@fluent/dedent/' imported from /Users/greg/dev/fluent.js/fluent-bundle/test/arguments_test.js
    at new NodeError (node:internal/errors:329:5)
    at legacyMainResolve (node:internal/modules/esm/resolve:272:9)
    at packageResolve (node:internal/modules/esm/resolve:707:14)
    at moduleResolve (node:internal/modules/esm/resolve:755:18)
    at Loader.defaultResolve [as _resolve] (node:internal/modules/esm/resolve:869:11)
    at Loader.resolve (node:internal/modules/esm/loader:88:40)
    at Loader.getModuleJob (node:internal/modules/esm/loader:241:28)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:57:40)
    at link (node:internal/modules/esm/module_job:56:36)
----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
----------|---------|----------|---------|---------|-------------------
All files |       0 |        0 |       0 |       0 |
----------|---------|----------|---------|---------|-------------------
```

Reply:
Suggestion: After running `npm dist` this now works. I feel that the command should be more robust.

Reply:
I also ran into this: https://github.com/projectfluent/fluent.js/issues/572

Reply:
Ok, I finally got the tests passing locally after commenting out the failing test.

Reply:
I was thinking while working on this that the tests ought to depend on the source rather than dist files. I left that change out of this PR because I felt it to be a reasonable future step.

Similarly the build-time tsc dependencies ought to be mapped to their dependencies' sources, but that too is a separate change.

Together, these should make individual scripts far less dependent on being executed in the "correct" order, and for the order of entries in the package.json `workspaces` to matter less.